src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Private/JavaScriptMarshalImpl.ToJs.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+
+using System.Runtime.CompilerServices;
+
+namespace System.Runtime.InteropServices.JavaScript.Private
+{
+    internal static unsafe partial class JavaScriptMarshalImpl
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static unsafe void DynamicMarshalToJs<T>(ref T value, ref object root, ref JavaScriptMarshallerArg arg, byte* extra, ref JavaScriptMarshallerArgSignature sig)
+        {
+            //log.Call(null, $"DynamicMarshalToJs {arg} at {(int)Unsafe.AsPointer(ref arg)}");
+            Type argType = typeof(T);
+            if (argType.IsPrimitive)
```

Comment:
the use of the temporary local may hinder various optimizations, it's worth checking into that at some point. the jit and linker both try to fold compile-time constants like this, but the linker might not be smart enough to do store-to-load forwarding on the local

Reply:
done